### ISSUE TITLE
v0.25 RC - QA API Keys

### ISSUE THREAD

**Issue by gmourier** [meilisearch/MeiliSearch#1979]
Hi folks 👋

After a first QA session, here are some feedbacks.

## `/keys` access

- The endpoints /keys are accessible without a master key defined for the instance, this API resource should not be accessible without it for security reasons.

## API Key Format 

- The formats of the `createdAt` and `updateAt` fields should be displayed with a precision in seconds (It facilitates reading while the microseconds precision is not needed here). For example, `createdAt: "2021-11-12T10:00:00Z"` instead of `"createdAt": "2021-12-07T14:30:22.633153Z"`.
- If an API key does not have a `description`, the field should be displayed with a `null` value.

## GET `/keys`

- API Key objects should be ordered by `createdAt` in `desc` order. (Most recent first)

## POST `/keys`

- `expiresAt` should be a mandatory field, `null` should be specified for no expiration date.
-  Setting `expiresAt` with `null` throw a validation error
- `expiresAt` if specified should throw an error if it's a past datetime from now. (This behavior is also found on the `patch` method).
-  Omitting the timezone part of the ISO-8601 format  at the end of the `expiresAt` field throws an error of format validation. An omitted timezone should default to `Z` automatically. e.g. Sending `"2021-11-12T10:00:00"`, result in `2021-11-12T10:00:00Z` (This behavior is also found on the `patch` method).

## DELETE `/keys/:key`

- Deleting a key returns a "Parse Error: The server returned a malformed response" on an HTTP client. (Postman in this case). The API key is however well deleted.

## Creating an index does not propagate the newly created index in the `indexes` array

When using this API Key:

```
{
        "description": "My API Key",
        "key": "AIcjX0aG56a0c73984fa2f2b7ac93eb3e33b1e227eb304aa8cfe89d3f25aba1f9c0c781b",
        "actions": [
            "*"
        ],
        "indexes": [
            "test"
        ],
        "expiresAt": "2021-12-07T15:05:00Z",
        "createdAt": "2021-12-07T15:10:42.861041Z",
        "updatedAt": "2021-12-07T17:43:26.936329Z"
}
```

I created an index `new_index` since I have the authorization given by `actions = *`. I expected to find it in the `indexes` array of the API Key. Also, access to the newly created index is not given.

```
{
        "description": "My API Key",
        "key": "AIcjX0aG56a0c73984fa2f2b7ac93eb3e33b1e227eb304aa8cfe89d3f25aba1f9c0c781b",
        "actions": [
            "*"
        ],
        "indexes": [
            "test",
            "newindex"
        ],
        "expiresAt": "2021-12-07T15:05:00Z",
        "createdAt": "2021-12-07T15:10:42.861041Z",
        "updatedAt": "2021-12-07T17:43:26.936329Z"
}
```

**Comment by gmourier** [comment]
Regarding 

**Creating an index does not propagate the newly created index in the indexes array**

When using this API Key:

{
        "description": "My API Key",
        "key": "AIcjX0aG56a0c73984fa2f2b7ac93eb3e33b1e227eb304aa8cfe89d3f25aba1f9c0c781b",
        "actions": [
            "*"
        ],
        "indexes": [
            "test"
        ],
        "expiresAt": "2021-12-07T15:05:00Z",
        "createdAt": "2021-12-07T15:10:42.861041Z",
        "updatedAt": "2021-12-07T17:43:26.936329Z"
}
I created an index new_index since I have the authorization given by actions = *. I expected to find it in the indexes array of the API Key. Also, access to the newly created index is not given.

{
        "description": "My API Key",
        "key": "AIcjX0aG56a0c73984fa2f2b7ac93eb3e33b1e227eb304aa8cfe89d3f25aba1f9c0c781b",
        "actions": [
            "*"
        ],
        "indexes": [
            "test",
            "newindex"
        ],
        "expiresAt": "2021-12-07T15:05:00Z",
        "createdAt": "2021-12-07T15:10:42.861041Z",
        "updatedAt": "2021-12-07T17:43:26.936329Z"
}

We decided to not handle it for now.

We are waiting for user feedback on this very particular case as we expect that the majority of indexes may be created with an admin key or for by key with `indexes=["*"]`.

**Comment by curquiza** [comment]
Closed by #1989
